src/libsync/syncengine.cpp

```diff
@@ -329,6 +329,51 @@ void SyncEngine::deleteStaleErrorBlacklistEntries(const SyncFileItemVector &sync
     _journal->deleteStaleErrorBlacklistEntries(blacklist_file_paths);
 }
 
+void SyncEngine::conflictRecordMaintenance()
+{
+    QByteArrayList conflictRecordPaths;
+
+    // Remove stale conflict entries from the database
+    if (account()->capabilities().uploadConflictFiles()) {
+        // If conflicts are uploaded, we just wipe all entries that no
+        // longer have a corresponding metadata entry.
```

Comment:
That sounds wrong to me,  what if the file failed to upload? It would not have a metadata, but the file would still exist on the disk.

Reply:
Very good point. Looks like I do need to check whether the actual files still exist here.